### DISCUSSION TITLE
How to preserve untracked files when in git repo?

### DISCUSSION THREAD

**Discussion by avoidik** [firecow/gitlab-ci-local#1326] [Q&A]
Hello,

It appears the tool takes whatever code was committed last. I would like to preserve untracked files. How can I achieve that?

Regards

**Comment by avoidik** [comment]
It seems this behavior was implemented on purpose

https://github.com/firecow/gitlab-ci-local/blob/82b8f88d241d931b908f48d6fd825d8493650b08/src/utils.ts#L254

```terminal
$ git ls-files -o --directory
```

this command here returns the list of files I'd like to keep

**Reply by firecow** [reply]
Yes, thats correct... 
It syncs tracked files to the gitlab build folders, and from there shell-executors runs jobs, and files are cp'ed to containers if docker-exectutor is used.
